flexmeasures/data/models/planning/utils.py

```diff
@@ -110,16 +111,43 @@ def get_prices(
             first_event_start = price_df.first_valid_index()
             last_event_end = price_df.last_valid_index() + resolution
             current_app.logger.warning(
-                f"Prices partially unknown for planning window. Trimming planning window (from {query_window[0]} until {query_window[-1]}) to {first_event_start} until {last_event_end}."
+                f"Prices partially unknown for planning window (sensor {price_sensor.id}). Trimming planning window (from {query_window[0]} until {query_window[-1]}) to {first_event_start} until {last_event_end}."
             )
             query_window = (first_event_start, last_event_end)
         else:
             raise UnknownPricesException(
-                "Prices partially unknown for planning window."
+                f"Prices partially unknown for planning window (sensor {price_sensor.id})."
             )
     return price_df, query_window
 
 
+def inflexible_device_forecasts(
```

Comment:
I feel the name of this function is not reflected in its code:

- inflexible device: a sensor ID is given, so this could be anything. 
- forecasts: As long as we are not passing a minimal horizon (`horizon_at_least`?), we don't seem to be searching for forecasts.

Reply:
> [...] if the requested schedule lies in the future, the data here will consist of (the most recent) forecasts (if any exist). And if the requested schedule lies in the past, the data here will consist of (the most recent) measurements (if any exist). The latter amounts to answering "What if we could have scheduled under perfect foresight?".

I have trouble finding a better term than "forecast". It seems like it is the most helpful term to quickly grasp what this function is supposed to be doing. But it should at least be very helpful if I add the above explanation to the docstring. I don't want to introduce a `horizon_at_least=timedelta(0)` filter, because it would destroy the use case of finding out what the best schedule would have been under perfect foresight.

For your first point: the function is passed a sensor rather than a sensor ID. Do you suggest I implement a check on whether the sensor is in fact a power sensor and/or belongs to the same asset as the to be scheduled sensor? If so, I feel it makes more sense to do that kind of validation as part of the marshmallow schema for the API endpoint and CLI command.

Reply:
We should not name a utility function according to what it is supposed to be doing, or what is should be used for, but simply what it is doing. I don't understand how this function is giving you forecasts when one doesn't filter by horizon.

The perfect-foresight cases seems to be an edge case, which could be solved via parametrisation (make horizon a parameter?).

The first point means that the name "inflexible" is not needed in the function's name, that's it.